File: findmeclinic/src/app/doctor.ts
import { Address } from './address';
import { DoctorAppointment } from './doctorappointment';

export class Doctor {
  name: string;
  role: string;
  gender: string;
  medicalLicense: string;
  profileImage: string;
  practiceStartedDate: string
  emailId: string;
  password: string;
  qualification: string
  specialization: string;
  clinicName: string;
  address: Address;
  phone: string;
  doctorAppointment: DoctorAppointment;
  clinicImage: string;


}

File: doctor-service/src/main/java/com/stackroute/domain/Doctor.java
package com.stackroute.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.List;

@Document
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Doctor
{

    @Id
    private String emailId;
    private String role;
    private String name;
    private String gender;
    private String phone;
    private String medicalLicense;
    private String profileImage;
    private String qualification;
    private String specialization;
    private Date practiceStartedDate;
    private String clinicName;
    private String clinicImage;
    private Address address;
    @Transient
    private String password;
    private Slot slot;
    private List<DoctorAppointment> doctorAppointmentList;

}
